signing-and-verification-workflow.md

```diff
@@ -0,0 +1,52 @@
+            1. Verify that the timestamp signing algorithm satisfies [algorithm requirements](./signature-specification.md#signature-algorithm-requirements).
+        1. Embed timestamp to the signature envelope.
+1. **Push the signature envelope:** Push the signature envelope generated in the previous step to the repository.
+1. **Generate signature artifact manifest:** As decribed in [signature specification](./signature-specification.md#storage) create the Notary v2 signature artifact manifest for the signature envelope generated in step 1.
+1. **Push signature artifact manifest:** Push Notary v2 signature artifact manifest to the repository.
+
+The user can push the OCI artifact to the repository before, after, or in between the signature generation process.
+
+## Verification workflow
+
+The user wants to pull OCI artifacts only if they are signed by the trusted publisher and the signature is valid.
+
+### Prerequisites
+
+- User has a digest of OCI artifact they want to download. If the user wants to use a tag to identify the artifact then that tag must be resolved to digest.
+- User has configured [trust store and trust policy](./trust-store-trust-policy-specification.md) required for signature verification.
+
+### Steps
+
+1. **Get signature artifact descriptors:** Using the ORAS [Manifest Referrers API](https://github.com/oras-project/artifacts-spec/blob/main/manifest-referrers-api.md) download  Notary v2 signature artifact descriptors. The ORAS referrers API only needs artifact digest and artifact type(optional) as input. The Notary v2 signature's artifact type is `application/vnd.cncf.notary.v2.signature`.
+1. **Get signature artifact manifests:** For each signature artifact descriptor, perform the following steps:
+    1. **Get signature artifact manifests:** Download the Notary v2 signature's artifact manifests for the given artifact descriptor.
+    1. **Filter signature artifact manifests:**
+        1. Based on the value of the `blobs->mediaType` attribute and supported signature envelope format, filter out signature manifests.
+        1. Depending upon the trust-store configuration, further filter out signature manifests. why? How? **TBD**(See [signature filtering](https://hackmd.io/@pritesh/signature_filtering)).
+    1. **Get and validate signatures:** On filtered Notary v2 signature artifact manifest, perform the following steps:
+        1. Download the signature envelope.
+        1. Validate the signature envelope using trust-store and trust-policy as mentioned in [signature evaluation](./trust-store-trust-policy-specification.md#signature-evaluation) section.
+        1. If signature validation fails, skip the below steps and move to the next signature artifact descriptor(step 2.1). If all signature artifact manifests have already been processed, fail the signature verification and exit.
```

Comment:
What constitutes a signature validation success/failure? Is this where we should call out checking for revocation lists, and/or timestamp validations?

Reply:
That's specified in previous step and the hyperlink points the signature evaluation process.

> Validate the signature envelope using trust-store and trust-policy as mentioned in [signature evaluation](./trust-store-trust-policy-specification.md#signature-evaluation) section